src/contracts/pods/EigenPodManager.sol

```diff
@@ -407,94 +236,73 @@ contract EigenPodManager is
         require(podOwner != address(0), "EigenPodManager._addShares: podOwner cannot be zero address");
-        require(shareAmount > 0, "EigenPodManager._addShares: amount must be greater than zero");
 
-        podOwnerShares[podOwner] += shareAmount;
+        uint256 currentShareDeficit = podOwnerShareDeficit[podOwner];
+
+        // skip dealing with deficit if there isn't any
+        if (currentShareDeficit == 0) {
+            podOwnerShares[podOwner] += shareAmount;
+            return shareAmount;            
+        }
 
-        // if the podOwner has delegated shares, record an increase in their delegated shares
-        if (!isInUndelegationLimbo(podOwner)) {
-            delegationManager.increaseDelegatedShares(podOwner, beaconChainETHStrategy, shareAmount);
+        // get rid of the whole deficit if possible
+        if (shareAmount >= currentShareDeficit) {
+            podOwnerShareDeficit[podOwner] = 0;
+            shareAmount -= currentShareDeficit;
+            podOwnerShares[podOwner] += shareAmount;
+            return shareAmount;
+        // otherwise get rid of as much deficit as possible
+        } else {
+            podOwnerShareDeficit[podOwner] -= shareAmount;
+            return 0;
         }
+
     }
 
-    // @notice Reduces the `podOwner`'s shares by `shareAmount` and performs a call to the DelegationManager to ensure delegated shares are also tracked correctly
-    function _removeShares(address podOwner, uint256 shareAmount) internal {
+    // @notice Reduces the `podOwner`'s shares by `shareAmount`, adding to deficit if necessary
+    // @dev Returns the number of shares removed from `podOwnerShares[podOwner]`
+    function _removeShares(address podOwner, uint256 shareAmount) internal returns (uint256) {
         require(podOwner != address(0), "EigenPodManager._removeShares: depositor cannot be zero address");
-        require(shareAmount != 0, "EigenPodManager._removeShares: shareAmount should not be zero!");
 
         uint256 currentPodOwnerShares = podOwnerShares[podOwner];
-        require(shareAmount <= currentPodOwnerShares, "EigenPodManager._removeShares: shareAmount too high");
-
-        unchecked {
-            currentPodOwnerShares = currentPodOwnerShares - shareAmount;
-        }
 
-        podOwnerShares[podOwner] = currentPodOwnerShares;
-
-        // if the podOwner has delegated shares, record a decrease in their delegated shares
-        if (!isInUndelegationLimbo(podOwner)) {
-            uint256[] memory shareAmounts = new uint256[](1);
-            shareAmounts[0] = shareAmount;
-            IStrategy[] memory strategies = new IStrategy[](1);
-            strategies[0] = beaconChainETHStrategy;
-            delegationManager.decreaseDelegatedShares(podOwner, strategies, shareAmounts);
+        // skip dealing with deficit if there isn't any need for it
+        if (shareAmount <= currentPodOwnerShares) {
+            podOwnerShares[podOwner] = currentPodOwnerShares - shareAmount;
```

Comment:
seems like the following invariant is true: `(deficit == 0 && podOwnerShares >= 0) || (deficit <= 0 && podOwnerShares == 0)`.

What if we just made `podOwnerShares` an `int256` and got rid of `deficit` entirely?

Reply:
yes, this is true, and the root issue of the previous deficit system was that it _wasn't_ always true.
I've tried to think through if using a signed integer would be much cleaner. thinking it'll make sense to do a quick PR to settle the debate / show everyone what it would look like.

Reply:
PR is here https://github.com/Layr-Labs/eigenlayer-contracts/pull/227
I recommend this change; I think it helps clarify how things are supposed to work.

Reply:
PR merged.